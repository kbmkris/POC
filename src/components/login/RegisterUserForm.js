import React from 'react';
import styles from '../../css/index.css';
import { PropTypes } from 'prop-types';

const RegisterUserForm = ({
  name,
  sapId,
  emailId,
  primarySkill,
  band,
  password,
  handleOnChange,
  handleOnClickSubmit,
  handleOnClickCancel
}) => (
  <div className="main-div">
    <div className="register-div">
      <h3> Register New User </h3>
      <form>
        <label
          className="label"
          htmlFor="name" >
          Name :
        </label>
        <input
          type="text"
          className="input"
          id="name"
          value={name}
          onChange={handleOnChange} />
        <br />
        <label
          className="label"
          htmlFor="sapId" >
          Sap Id :
        </label>
        <input
          type="number"
          className="input"
          id="sapId"
          value={sapId}
          onChange={handleOnChange} />
        <br />
        <label
          className="label"
          htmlFor="emailId" >
          Email Id :
        </label>
        <input
          type="email"
          className="input"
          id="emailId"
          value={emailId}
          onChange={handleOnChange} />
        <br />
        <label
          className="label"
          htmlFor="band" >
          Band :
        </label>
        <input
          type="text"
          className="input"
          id="band"
          value={band}
          onChange={handleOnChange} />
        <br />
        <label
          className="label"
          htmlFor="primarySkill" >
          Primary Skill :
        </label>
        <input
          type="text"
          className="input"
          id="primarySkill"
          value={primarySkill}
          onChange={handleOnChange} />
        <br />
        <label
          className="label"
          htmlFor="password" >
          Password :
        </label>
        <input
          type="password"
          className="input"
          id="password"
          value={password}
          onChange={handleOnChange} />
        <br />
        <div className="button-div">
          <button
            className="button"
            onClick={handleOnClickSubmit} >
            Submit
          </button>
          <button
            className="button"
            onClick={handleOnClickCancel} >
            Cancel
          </button>
        </div>
      </form>
    </div>
  </div>
);


RegisterUserForm.propTypes = {
  name : PropTypes.string.isRequired,
  sapId : PropTypes.number.isRequired,
  emailId : PropTypes.string.isRequired,
  primarySkill : PropTypes.string.isRequired,
  band : PropTypes.string.isRequired,
  password : PropTypes.string.isRequired,
  handleOnChange : PropTypes.func.isRequired,
  handleOnClickSubmit : PropTypes.func.isRequired,
  handleOnClickCancel : PropTypes.func.isRequired
};

export default RegisterUserForm;
