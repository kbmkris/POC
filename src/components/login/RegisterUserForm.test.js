import React from "react";
import RegisterUserForm from "./RegisterUserForm";
import renderer from "react-test-renderer";
import { mount, shallow } from "enzyme";

describe("RegisterUserForm Test", () => {
  it("Snapshot - component should render", () => {
    const handleOnChange = jest.fn();
    const handleOnClickSubmit = jest.fn();
    const handleOnClickCancel = jest.fn();
    const component = renderer
      .create(
        <RegisterUserForm
          name={""}
          sapId={0}
          emailId={""}
          primarySkill={""}
          band={""}
          password={""}
          isLoading={false}
          handleOnChange={handleOnChange}
          handleOnClickSubmit={handleOnClickSubmit}
          handleOnClickCancel={handleOnClickCancel}
        />,
      );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe ("Dom validation using enzyme", () => {
  function setup (isLoading) {
    const props = {
      name: "",
      sapId: 0,
      emailId: "",
      primarySkill: "",
      band: "",
      password: "",
      isLoading: isLoading,
      handleOnChange: jest.fn(),
      handleOnClickSubmit: jest.fn(),
      handleOnClickCancel: jest.fn()
    };
    return shallow(<RegisterUserForm {...props} />);
  }
  it("check for all elements inside RegisterUserForm", () => {
    const wrapper = setup(false);
    expect(wrapper.find("form").length).toBe(1);
    expect(wrapper.find("h3").text()).toEqual(" Register New User ");
    expect(wrapper.find("input").length).toBe(6);
    expect(wrapper.find("button").length).toBe(2);
    expect(wrapper.find("#name").length).toBe(1);
    expect(wrapper.find("#sapId").length).toBe(1);
    expect(wrapper.find("#emailId").length).toBe(1);
    expect(wrapper.find("#band").length).toBe(1);
    expect(wrapper.find("#primarySkill").length).toBe(1);
    expect(wrapper.find("#password").length).toBe(1);
    expect(wrapper.find("button").at(0).text()).toEqual("Submit");
    expect(wrapper.find("button").at(1).text()).toEqual("Cancel");
  });
  it("check for submit button value changes when loading", () => {
    const wrapper = setup(true);
    expect(wrapper.find("button").at(0).text()).toEqual("Registering...");
   });
});
