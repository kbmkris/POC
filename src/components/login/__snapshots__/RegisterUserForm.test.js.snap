// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterUserForm Test Snapshot - component should render 1`] = `
<div
  className="main-div"
>
  <div
    className="register-div"
  >
    <h3>
       Register New User 
    </h3>
    <form>
      <label
        className="label"
        htmlFor="name"
      >
        Name :
      </label>
      <input
        className="input"
        id="name"
        onChange={[MockFunction]}
        type="text"
        value=""
      />
      <br />
      <label
        className="label"
        htmlFor="sapId"
      >
        Sap Id :
      </label>
      <input
        className="input"
        id="sapId"
        onChange={[MockFunction]}
        type="number"
        value={0}
      />
      <br />
      <label
        className="label"
        htmlFor="emailId"
      >
        Email Id :
      </label>
      <input
        className="input"
        id="emailId"
        onChange={[MockFunction]}
        type="email"
        value=""
      />
      <br />
      <label
        className="label"
        htmlFor="band"
      >
        Band :
      </label>
      <input
        className="input"
        id="band"
        onChange={[MockFunction]}
        type="text"
        value=""
      />
      <br />
      <label
        className="label"
        htmlFor="primarySkill"
      >
        Primary Skill :
      </label>
      <input
        className="input"
        id="primarySkill"
        onChange={[MockFunction]}
        type="text"
        value=""
      />
      <br />
      <label
        className="label"
        htmlFor="password"
      >
        Password :
      </label>
      <input
        className="input"
        id="password"
        onChange={[MockFunction]}
        type="password"
        value=""
      />
      <br />
      <div
        className="button-div"
      >
        <button
          className="button"
          name="submitButton"
          onClick={[MockFunction]}
        >
          Submit
        </button>
        <button
          className="button"
          name="cancelButton"
          onClick={[MockFunction]}
        >
          Cancel
        </button>
      </div>
    </form>
  </div>
</div>
`;
